romancal/assign_wcs/assign_wcs_step.py

```diff
@@ -141,17 +142,18 @@ def wfi_distortion(model, reference_files):
     transform = dist.coordinate_distortion_transform
 
     try:
-        bbox = transform.bounding_box
+        bbox = transform.bounding_box.bounding_box(order="F")
     except NotImplementedError:
         # Check if the transform in the reference file has a ``bounding_box``.
         # If not set a ``bounding_box`` equal to the size of the image after
         # assembling all distortion corrections.
         bbox = None
     dist.close()
 
-    if bbox is None:
-        transform.bounding_box = wcs_bbox_from_shape(model.data.shape)
-    else:
-        transform.bounding_box = bbox
+    bind_bounding_box(
+        transform,
+        wcs_bbox_from_shape(model.data.shape) if bbox is None else bbox,
+        order="F",
+    )
```

Comment:
This is a case where the GWCS warning identified a bug. It just did not get picked up by any tests...

Note https://github.com/spacetelescope/romancal/blob/281aa556a388363f98ed8a69ad1de46ef4b78297/romancal/assign_wcs/utils.py#L20-L35

Reports to create a bounding box in the `x, y` ordering **but** the `transform.bounding_box =` was making the assumption this was a `y, x` box.

With this change it is now being assigned correctly with an `x, y` ordering.